shell/platform/android/io/flutter/plugin/platform/ImageReaderPlatformViewRenderTarget.java

```diff
@@ -0,0 +1,109 @@
+package io.flutter.plugin.platform;
+
+import android.graphics.Canvas;
+import android.hardware.HardwareBuffer;
+import android.media.Image;
+import android.media.ImageReader;
+import android.os.Handler;
+import android.view.Surface;
+import io.flutter.view.TextureRegistry.ImageTextureEntry;
+
+public class ImageReaderPlatformViewRenderTarget implements PlatformViewRenderTarget {
+  private ImageTextureEntry textureEntry;
+  private boolean mustRecreateImageReader = false;
+  private ImageReader reader;
+  private int bufferWidth = 0;
+  private int bufferHeight = 0;
+  private static final String TAG = "ImageReaderPlatformViewRenderTarget";
+
+  private void closeReader() {
+    if (this.reader != null) {
+      this.reader.close();
+      this.reader = null;
+    }
+  }
+
+  private void recreateImageReaderIfNeeded() {
+    if (!mustRecreateImageReader) {
+      return;
+    }
+    mustRecreateImageReader = false;
+    closeReader();
+    this.reader = createImageReader();
+  }
+
+  private final Handler onImageAvailableHandler = new Handler();
+  private final ImageReader.OnImageAvailableListener onImageAvailableListener =
+      new ImageReader.OnImageAvailableListener() {
+        @Override
+        public void onImageAvailable(ImageReader reader) {
+          final Image image = reader.acquireLatestImage();
+          if (image == null) {
+            return;
+          }
+          textureEntry.pushImage(image);
+        }
+      };
+
+  protected ImageReader createImageReader() {
+    final ImageReader.Builder builder = new ImageReader.Builder(bufferWidth, bufferHeight);
+    // Allow for double buffering.
+    builder.setMaxImages(2);
+    // Hint that images will only be touched by GPU.
+    builder.setUsage(
```

Comment:
We are never going to write to the image as either as an attachment or as a shader resource. Only USAGE_GPU_SAMPLED_IMAGE should suffice.

Reply:
The images will be written to when we render the Android Platform View into them. Afterwards you are correct that they will only be read by the engine.